### PR TITLE
Stability doc + new feature: support for OAuth

### PR DESCRIPTION
**Pull request by a-akimov**
> [!NOTE]
> **Medium Risk**
> Adds an interactive OAuth path that handles tokens and opens a local callback listener; behavior is optional and well-tested, but it touches authentication and user-facing security semantics.
> 
> **Overview**
> **Interactive OAuth for full audits** — New `--oauth` runs authorization-code + PKCE in the browser: discover the AS (RFC 9728/8414), optionally register dynamically (RFC 7591) or use `--client-id`, catch the redirect on `127.0.0.1` via `--callback-port`, then audit with an in-memory `Authorization: Bearer` token (same path as `--token`). CLI validates conflicts (no `--token`/env credential), HTTP-only targets, and failed flows exit **1**.
> 
> **Stability contract** — New `docs/stability.mdx` (nav + CHANGELOG) documents stable surfaces (`rule_id`, `schema_version`, CLI/exit codes, no credentials in logs/reports) vs evolving score/messages; exit **1** now explicitly includes failed `--oauth`.
> 
> **Tests** — Broad `tests/test_oauth.py` and `TestOAuthCliFlow` in `tests/test_cli.py` cover the loopback edge cases and CLI wiring.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 3c622ae39d577ed0f468b1a2c4252658036be991. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: mcpscore/oauth.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+        self._result: asyncio.Future[dict[str, str]] = asyncio.get_event_loop().create_future()
+        self.port: int = 0
+
+    @property
+    def redirect_uri(self) -> str:
+        return f"http://127.0.0.1:{self.port}{CALLBACK_PATH}"
+
+    async def start(self) -> None:
+        self._server = await asyncio.start_server(self._handle, host="127.0.0.1", port=0)
+        self.port = self._server.sockets[0].getsockname()[1]
```
[the remark this answers is not in the document]
Comment (a-akimov) [reply mcpscore/oauth.py:103]: pyright passes this file with 0 errors at our gate settings; current typeshed types Server.sockets as a non-optional tuple

File: mcpscore/oauth.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+        storage = _MemoryTokenStorage()
+        if client_id is not None:
+            # Pre-seeding client info makes the SDK skip dynamic registration —
+            # the escape hatch for authorization servers without RFC 7591.
+            await storage.set_client_info(
+                OAuthClientInformationFull(
+                    client_id=client_id,
+                    redirect_uris=[AnyUrl(callback.redirect_uri)],
+                    token_endpoint_auth_method="none",  # noqa: S106 — public-client method name, not a secret
+                )
```
[the remark this answers is not in the document]
Comment (a-akimov) [reply mcpscore/oauth.py:196]: Refuted empirically: AnyUrl(...) direct construction runs in every OAuth test — 509 passing on pydantic 2.13.4. Pydantic v2's Url types support validated direct construction. Passing plain strings is what our pyright gate rejected two rounds ago; AnyUrl was added precisely to satisfy it. The import therefore stays.

File: mcpscore/oauth.py [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+import asyncio
+import logging
+from typing import TYPE_CHECKING
+from urllib.parse import parse_qs, urlsplit
+import webbrowser
+
+import httpx2
+from mcp.client.auth import OAuthClientProvider, OAuthRegistrationError, OAuthTokenError
+from mcp.shared.auth import AuthorizationCodeResult, OAuthClientInformationFull, OAuthClientMetadata, OAuthToken
+from pydantic import AnyUrl
```
[the remark this answers is not in the document]
Comment (a-akimov) [reply mcpscore/oauth.py:29]: Refuted empirically: AnyUrl(...) direct construction runs in every OAuth test — 509 passing on pydantic 2.13.4. Pydantic v2's Url types support validated direct construction. Passing plain strings is what our pyright gate rejected two rounds ago; AnyUrl was added precisely to satisfy it. The import therefore stays.
